File: frontend/src/app/models/client/client-view-dto.ts
import { TrainerViewDTO } from "../trainer/trainer-view-dto";

export interface ClientViewDTO {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    imageUrl: string | null;
    goal: string | null;
    birthday: string;
    initialWeight: number | null;
    initialHeight: number | null;
    personalTrainer: TrainerViewDTO;
  }
  

File: backend/PulseGym.Logic/DTO/ClientDTO/ClientViewDTO.cs
namespace PulseGym.Logic.DTO
{
    public class ClientViewDTO
    {
        public Guid Id { get; set; }

        public required string FirstName { get; set; }

        public required string LastName { get; set; }

        public required string Email { get; set; }

        public required DateTime Birthday { get; set; }

        public string? ImageUrl { get; set; }

        public string? Goal { get; set; }

        public double? InitialWeight { get; set; }

        public double? InitialHeight { get; set; }

        public TrainerViewDTO? PersonalTrainer { get; set; }

        public ICollection<MembershipProgramViewDTO>? MembershipPrograms { get; set; }
    }
}
